kms/key/Acornfile

```diff
@@ -0,0 +1,135 @@
+	keySpec: "SYMMETRIC_DEFAULT"
+	// The usage for the Key. Options are "ENCRYPT_DECRYPT", "SIGN_VERIFY", and "GENERATE_VERIFY_HMAC". Each keySpec is only compatible with certain keyUsages - see README for more info. Default is "ENCRYPT_DECRYPT".
+	keyUsage: "ENCRYPT_DECRYPT"
+	// The time (in days) that must pass after key deletion is requested before the key is deleted. Default is 30. Minimum is 7. Maximum is 30.
+	pendingWindowDays: 30
+	// AWS IAM policy to attach to the Key. Optional.
+	keyPolicy: {}
+}
+
+services: key: {
+	name: "AWS KMS Key"
+	generated: job: "apply"
+	consumer: permissions: rules: [{
+		apiGroups: ["aws.acorn.io"]
+		verbs: [
+			"kms:Decrypt",
+			"kms:DescribeKey",
+			"kms:Encrypt",
+			"kms:GenerateDataKey",
+			"kms:GenerateDataKeyPair",
+			"kms:GenerateDataKeyPairWithoutPlaintext",
+			"kms:GenerateMac",
+			"kms:GenerateRandom",
+			"kms:GetKeyPolicy",
+			"kms:GetKeyRotationStatus",
+			"kms:GetPublicKey",
+			"kms:ListAliases",
+			"kms:ListGrants",
+			"kms:ListKeyPolicies",
+			"kms:ListResourceTags",
+			"kms:ListRetirableGrants",
+			"kms:ReEncryptFrom",
+			"kms:ReEncryptTo",
+			"kms:Sign",
+			"kms:Verify",
+			"kms:VerifyMac",
+		]
+		resources: ["*"]
+	}]
+}
+
+jobs: apply: {
+	build: {
+		context: "."
+		additionalContexts: {
+			common: "../../libs"
+			utils:  "../../utils"
+		}
+	}
+	files: "/app/config.json": std.toJSON(args)
+	env: {
+		CDK_DEFAULT_ACCOUNT: "@{secrets.aws-context.account-id}"
+		CDK_DEFAULT_REGION:  "@{secrets.aws-context.aws-region}"
+		VPC_ID:              "@{secrets.aws-context.vpc-id}"
+		ACORN_ACCOUNT:       "@{acorn.account}"
+		ACORN_NAME:          "@{acorn.name}"
+		ACORN_PROJECT:       "@{acorn.project}"
+		ACORN_EXTERNAL_ID:   "@{acorn.externalId}"
+	}
+	events: ["create", "update", "delete"]
+	permissions: rules: [{
+		apiGroup: "aws.acorn.io"
+		verbs: [
+			"cloudformation:DescribeStacks",
+			"cloudformation:CreateChangeSet",
+			"cloudformation:DescribeChangeSet",
+			"cloudformation:DescribeStackEvents",
+			"cloudformation:DescribeStackResources",
+			"cloudformation:ExecuteChangeSet",
+			"cloudformation:PreviewStackUpdate",
+			"cloudformation:UpdateStack",
+			"cloudformation:GetTemplateSummary",
+			"cloudformation:DeleteStack",
+			"kms:*",
+		]
+		resources: ["*"]
+	}, {
+		apiGroup: "aws.acorn.io"
+		verbs: [
+			"ec2:DescribeAvailabilityZones",
+			"ec2:DescribeVpcs",
+			"ec2:DescribeSubnets",
+			"ec2:DescribeRouteTables",
+		]
+		resources: ["*"]
+	}, {
+		apiGroup: "api.acorn.io"
+		verbs: [
+			"create",
+		]
+		resources: ["events"]
+	}]
+}
+
+secrets: "aws-context": {
+	name:     "AWS Context"
+	external: "context://aws"
+	type:     "opaque"
+	data: {
+		"account-id": ""
+		"vpc-id":     ""
+		"aws-region": ""
+	}
+}
+
+localData: info: """
```

Comment:
This should tell the user how to consume it. So where are the values going to be located...

arn is accessed by ${services.key.data.arn} is good, but it should also let the user know where to get any other values.

@sosodev should this use the @{@{service}. notation?

Reply:
When rendering an example of how to use it yes. When rendering actual information no.